src/Symfony/Bridge/PhpUnit/ErrorAssert.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Symfony\Bridge\PhpUnit;
+
+/**
+ * Test that your code triggers expected error messages.
+ *
+ * @author Christian Flothmann <christian.flothmann@xabbuh.de>
+ */
+final class ErrorAssert
+{
+    public static function assertDeprecationsAreTriggered($expectedMessages, callable $testCode)
+    {
+        self::assertErrorsAreTriggered(E_USER_DEPRECATED, $expectedMessages, $testCode);
+    }
+
+    public static function assertErrorsAreTriggered($expectedType, $expectedMessages, callable $testCode)
+    {
+        $triggeredMessages = array();
+
+        set_error_handler(function ($type, $message) use ($expectedType, &$triggeredMessages) {
+            if ($expectedType !== $type) {
+                restore_error_handler();
+
+                return call_user_func_array('PHPUnit_Util_ErrorHandler::handleError', func_get_args());
+            }
+
+            $triggeredMessages[] = $message;
+        });
+
+        $testCode();
```

Comment:
What about putting the entire function in a try/finally structure in case an exception is thrown ?
```php
try {
    // ...
} finally {
   restore_error_handler();
}